docs/DEVELOPING.md

```diff
@@ -124,6 +127,227 @@ Take a look at the [MongoDB connector](../connectors/sources/mongo.py) for more
+  - Increase your page sizes
+  - Cache records in-memory rather than looking them up multiple times
+2. Respect the rate limit errors
+  - handle (don't crash on) 429 errors
+  - read the `Retry-After` header
+  - know if the API uses mechanisms other than 429 codes and `Retry-After`
+
+For Sharepoint Online, there are two important limits for the Sharepoint API:
+
+- Per minute app can consume between 1,200 to 6,000 Resource Units.
+- Per day app can consume between 1,200,000 to 6,000,000 Resource Units.
+
+These quotas are shared between Graph API and Sharepoint REST API - the latter consumes more resource units.
+
+If the quotas are exceeded, some endpoints return 503 responses which contain the `Retry-After` header, and are used instead of 429 codes.
+
+Putting all this together, we know that we must specially handle 503 and 429 responses, looking for `Retry-After` headers,
+and design our connector knowing that we must make the absolute most out of each request, since we will nearly always run up on the per-minute and per-day limits.
+
+##### What is the API's default page size?
+
+Related to the above section, many developers are surprised at how small default page sizes may be.
+If your goal is to reduce the number of requests you make to a 3rd-party, page sizes of 10 or 20 rarely make sense.
+
+However, page sizes of 1,000,000,000,000 likely aren't a good idea either, no matter how small a given record is.
+It's important to consider how large of a response payload you can reasonably handle at once, and craft your page size accordingly.
+
+Sharepoint Online has a default page size of 100.
+For smaller-payload resources like Users and Groups, this can probably be safely increased by an order of magnitude or more.
+
+##### When should failure responses be retried?
+
+As discussed above, many APIs utilize rate limits and will issue 429 responses for requests that should be retried later.
+However, sometimes there are other cases that warrant retries. 
+
+Some services experience frequent-but-transient timeout issues.
+Some services experiencing a large volume of write operations may sometimes respond with 409 "conflict" codes.
+Sometimes downloading large files encounter packet loss, and while the response is a 200 OK, the checksum of the downloaded file does not match.
+
+All of these are operations that may make sense to retry a time or two.
+
+A note on retries - do not retry infinitely, and use backoff.
+A service may block your connector or specific features if you retry too aggressively.
+
+##### What is the permission model used by the 3rd-party?
+
+Whether or not you plan to support Document Level Security with your connector, it is important to understand how the 3rd-party system protects data and restricts access.
+This allows you to ensure that synced documents contain the relevant metadata to indicate ownership/permissions so that any consumer of your connector can choose to implement their own DLS if they so desire.
+
```

Comment:
Good point 💯

Reply:
I stole this content from a guide that @artem-shelkovnikov is working on. I thought it might make sense to have consistent examples here, but I don't see an issue with presenting the same info in a sharepoint-specific doc.